### ISSUE TITLE
Statistics and admin

### ISSUE THREAD

**Issue by leotiger** [getgrav/grav-plugin-admin#927]
Woops, another one:

The statistics panel counts admin hits. Shouldn't there be at least an option to exclude admin interactions?

Here's a code proposal for the trackHit function inside popularity.php:

```
    public function trackHit()
    {
        // Don't track bot or crawler requests
        if (!Grav::instance()['browser']->isHuman()) {
            return;
        }
        // Exclude tracking while inside admin
	if ($this->isAdmin()) {
		return;
	}

        /** @var Page $page */
        $page = Grav::instance()['page'];
        $relative_url = str_replace(Grav::instance()['base_url_relative'], '', $page->url());

        // Don't track error pages or pages that have no route
        if ($page->template() == 'error' || !$page->route()) {
            return;
        }

        // Make sure no 'widcard-style' ignore matches this url
        foreach ((array)$this->config->get('plugins.admin.popularity.ignore') as $ignore) {
            if (fnmatch($ignore, $relative_url)) {
                return;
            }
        }

        // initial creation if it doesn't exist
        if (!file_exists($this->data_path)) {
            mkdir($this->data_path);
            $this->flushPopularity();
        }

        // Update the data we want to track
        $this->updateDaily();
        $this->updateMonthly();
        $this->updateTotals($page->route());
        $this->updateVisitors(Grav::instance()['uri']->ip());

    }
```

**Comment by leotiger** [comment]
Update on this: 
Defining a custom error page for 404 errors in the error plugin configuration requires that the page you use as an error page has a template called "error" associated, otherwise 404 errors, even if they occur inside admin without any visual manifestation, will increment the hits and thus statistics displayed in the admin dashboard.

**Comment by flaviocopes** [comment]
The reason is the Popularity functionality (the ones that generates the Admin stats) does this check: `$page->template() == 'error'`. So your custom error page must have a 'error' template otherwise it's going to be counted as a regular page. It's not a bug but a (silent?) convention.

**Comment by leotiger** [comment]
Little bit to silent :-) 

I would say that the error plugin needs a configuration option, but that's not the place to ask for that. As of the admin plugin, for sure it's something that has to be addressed one way or the other... to show statistics and to see them fly, up up to the sky :-), with each click on dashboard is something that makes a bit "nervous".

**Comment by leotiger** [comment]
To improve the behavior, for me it's a behavior bug: How about implementing a configuration option that allows to define templates that won't trigger hits inside popularity? I feel that this would help people (it's a clear hint as well) and enhances the platform.
Regards.
uh

**Comment by flaviocopes** [comment]
I just documented it: https://github.com/getgrav/grav-plugin-error/commit/1cc37ccc5a73ebd2cb8fe0bff84a34b2478b1cce

It's a matter of conventions, if you specify a custom error page, it should have a "error" template. Do you have a specific reason why you can't use that?

Yes we could have an option for this, but we have to put a line on what's configurable and what's no, and so a balance on the number of configuration options offered before it gets overwhelming. Just my opinion here, not from the whole Grav team, so other views are welcome.

**Comment by leotiger** [comment]
Thanks for looking into this. Maybe I didn't provide enough explanation. Let's have a closer look into what popularity does:

It does not track hits on pages that use a template with the name error, neither in front end nor in admin. 

My point is that popularity should never track hits while in admin (Grav may add an option to allow hit tracking while in admin, but generally site administrators prefer to not include their own hits) and if things work that way errors in front end may be tracked or not, simply depending on the template associated to the error page. This way totals.json file inside the log/popularity directory can include hits related with 404 errors. The only thing is not to use the "error" template for the error page.

Short answer: I can live with how it works now and with a hint inside the documentation other users may avoid seeing the statistics widget going crazy inside admin dashboard.

**Comment by rhukster** [comment]
Popularity should not be tracking any admin activity.  Also you can add extra paths that it will ignore via the admin plugins configuration YAML.

**Comment by leotiger** [comment]
Exactly, it should not but it does under certain conditions! A 404 error related with admin activity (testing, code stuff, etc.) may provoke this, can produce 404 errors while in admin. They won't show up in the interface necessarily but they produce hits that impact popularity.

**Comment by leotiger** [comment]
A possible solution: while admin is active, GRAV could attach by force an internal error template that is excluded from tracking in the error plugin. This seems an easy solution for this which won't mean a lot of code changes.

**Comment by flaviocopes** [comment]
👍  it does not track Admin activity, the problem here was that you had a 404 error, the custom 404 error page was not of `error` page type, and the logic of Popularity is told to just not track `error` page types hits:

https://github.com/getgrav/grav-plugin-admin/blob/develop/classes/popularity.php#L58-L61
```php
        // Don't track error pages or pages that have no route
        if ($page->template() == 'error' || !$page->route()) {
            return;
        }
```

Since your 404 page was not a `error` page, it was counted like any other page.

Admin provides its own error page, but for some reason the frontend error page is also looked at, in the process. Worth investigating.
